syntax = "proto3";
package qr;

service qr {
  // client streaming: клиент шлёт последовательность PhotoChunk, сервер в конце возвращает UploadSummary
  rpc UploadPhotos (stream PhotoChunk) returns (UploadSummary);
}

message PhotoChunk {
  string user_id = 1;
  string username = 2;
  string filename = 3;
  string disk = 4;
  string message_id = 5;
  int32 chunk_index = 6;
  bool is_last_chunk = 7;
  bytes data = 8;
}

message UploadSummary {
  bool confirmed = 1;
  string nickname = 2;
  string username = 3;
  string disk = 4;
  string datetime = 5;
}
